mempool/reactor.go

```diff
@@ -20,16 +21,26 @@ import (
 type Reactor struct {
 	p2p.BaseReactor
 	config  *cfg.MempoolConfig
-	mempool *CListMempool
+	mempool Mempool
 	ids     *mempoolIDs
+
```

Comment:
`mempoolIDs` is used to assign a unique identifier of type `uint16` to each peer (identified by `p2p.ID`). These ids are only known locally to the node, and only used to keep track of the senders of each transaction. 

Digging a bit on the history of this code I found [this PR](https://github.com/tendermint/tendermint/pull/2778) that introduces the data structure. It says: "Instead of using the 20byte peer ID, it instead uses a local map from peerID to uint16 counter, so every peer adds 2 bytes. (Word aligned to probably make it 8 bytes)".

Now that I think about it, it's not worth to keep all this code just to save 12 bytes per peer.

Reply:
💯

Reply:
Tracked in new issue #1146.